src/components/app.tsx

```diff
@@ -5,13 +5,41 @@ import { DragDropContext } from 'react-dnd';
 import HTML5Backend from 'react-dnd-html5-backend';
 import * as SplitPane from 'react-split-pane';
 
+import { Dispatch } from 'redux';
+import { StateWithHistory } from 'redux-undo';
+import {Data, InlineData, isInlineData, isUrlData, UrlData} from 'vega-lite/build/src/data';
+import { datasetUrlLoad } from '../actions';
+import { StateBase } from '../models/index';
+
 import {DataPane} from './data-pane';
 import {EncodingPane} from './encoding-pane';
 import {Header} from './header';
 import {ViewPane} from './view-pane';
 
 
-class AppBase extends React.PureComponent<any, {}> {
+export type VoyagerConfig = Object;
+export type VoyagerData = Data;
+
+interface Props extends React.Props<AppBase> {
+  config?: VoyagerConfig;
+  data?: Data;
+  dispatch: Dispatch<StateWithHistory<Readonly<StateBase>>>;
+}
+
+class AppBase extends React.PureComponent<Props, {}> {
+
+  constructor(props: any) {
+    super(props);
+  }
+
+  public componentWillUpdate(nextProps: Props) {
+    this.update(nextProps);
```

Comment:
Wondering if we have to compare with the previous state to avoid unnecessary load request?